app/mailers/user_mailer.rb

```diff
@@ -8,4 +8,11 @@ def new_follower_notification(follower, user)
     mail(to: "#{user.name} <#{user.email}>",
          subject: 'You have a new follower on Librarium')
   end
+
+  def weekly_activity_summary(recipient_email, activity_array)
```

Comment:
The activity_array should be calculated inside this method, not received as parameter. It's better that way, because between the time you call this method and the moment it actually executes could take a while, in the meantime the activity_array is stored in memory and the object can be quite large. Better to query it at the precise moment that it's actually used